### PR TITLE
New Update: Mathematical Equations in mainPanel for SIRS and other models 

### PR DESCRIPTION
**Pull request by kle6951**
**_Features were implemented:_** 
**1** I finished adding the mathematical equations for the SIRS model in the main panel (ie: Write the necessary LaTeX generating function in R/model_LaTeX.R) addressed by Bryce's notes in #17 
**2** I updated the section `Reproductive ratio` by using the curly bracket for the conditions as Ashok has mentioned before 

**_There are a few things needed to consider:_**
1. Please double-check the mathematical equation for the reproductive ratio of the SIRS 
2. In the SIRS model, based on the equations provided by Ashok, I left the population as `N` instead of `N^q` in other models. Please let me know which one is more preferable.
3. Please double-check the Greek symbol as `\nu` mentioned in my comments in the code

### CODE REVIEW

**Comment by ashokkrish** [comment]
We have to have a separate meeting to derive and print the **basic reproduction number (_R_0)** for each model. I will assign a specific set of articles/book chapters that helps us with the derivation. This will involve inverting and multiplying matrices and then eventually calculating the largest eigenvalue. More details to follow.

**Comment by ashokkrish** [comment]
I would like the ODEs rendered without this left border text

image

Let's left indent all the equations. For example, an SIR model without vital dynamics under the pseudo mass action formulation would look like this

image

Another example, an SEIRD model under the true mass action formulation would like this

image

File: R/model_LaTeX.R [hunk 1]
```diff
@@ -2,21 +2,17 @@ SIR_LaTeX <- function(mu) {
...
       r"(Susceptible $$\frac{dS}{dt} = - \beta \frac{ S I}{N^q}$$)",
-      r"(Infectious $$\frac{dI}{dt} = \frac{\beta S I}{N^q} - \gamma I $$)",
-      r"(Recovered $$\frac{dR}{dt} = \gamma I $$)",
+      r"(Infectious $$\frac{dI}{dt} = \frac{\beta S I}{N^q} - \gamma {I} $$)",
+      r"(Recovered $$\frac{dR}{dt} = \gamma {I} $$)",
       r"(Reproductive ratio $$R_0 =  \frac{\beta}{\gamma} S(0)^q$$)",
-      "q-Value", br(),
-      r"($$ 1, frequency-dependent $$)", br(),
-      r"($$ 0, density-dependent $$)"
+      "q-Value $$\\begin{equation} x = \\begin{cases} 1 & \\text{, } frequency − dependent \\\\ 0 & \\text{, } density − dependent \\end{cases}\\end{equation} $$"
```
Comment (bryce-carson) [inline R/model_LaTeX.R:8]: Please use raw strings when writing $\LaTeX{}$. Inside the raw string you will only have to write $\LaTeX{}$, and won't need to escape backslash characters which are used to call macros and commands.

File: R/model_LaTeX.R [hunk 2]
```diff
@@ -2,21 +2,17 @@ SIR_LaTeX <- function(mu) {
...
       r"(Susceptible $$\frac{dS}{dt} =\mu_B N - \mu_D S - \beta \frac{ S I}{N^q}$$)",
-      r"(Infectious $$\frac{dI}{dt} = \frac{\beta S I}{N^q} - \gamma I - \mu_D I$$)",
-      r"(Recovered $$\frac{dR}{dt} = \gamma I - \mu_D R $$)",
+      r"(Infectious $$\frac{dI}{dt} = \frac{\beta S I}{N^q} - \gamma {I} - \mu_D I$$)",
+      r"(Recovered $$\frac{dR}{dt} = \gamma {I} - \mu_D R $$)",
       r"(Reproductive ratio $$R_0 =  \frac{\beta}{\gamma} S(0)^q$$)",
-      "q-Value", br(),
-      r"($$ 1, frequency-dependent $$)", br(),
-      r"($$ 0, density-dependent $$)"
+      "q-Value $$\\begin{equation} x = \\begin{cases} 1 & \\text{, } frequency − dependent \\\\ 0 & \\text{, } density − dependent \\end{cases}\\end{equation} $$"
```
Comment (bryce-carson) [inline R/model_LaTeX.R:15]: Please use raw strings when writing $\LaTeX{}$. Inside the raw string you will only have to write $\LaTeX{}$, and won't need to escape backslash characters which are used to call macros and commands.

File: R/model_LaTeX.R [hunk 3]
```diff
@@ -26,25 +22,43 @@ SIR_LaTeX <- function(mu) {
...
 ## server.R pass input$muValue also.
-SIRS_LaTeX <- function() {}
+SIRS_LaTeX <- function(mu) {
+  switch(mu + 1, 
+         helpText(
+           r"(Susceptible $$\frac{dS}{dt} = - \beta \frac{ S I}{N} + \xi{R}$$)",
+           r"(Infectious $$\frac{dI}{dt} = \frac{\beta S I}{N} - \gamma {I}$$)",
+           r"(Recovered $$\frac{dR}{dt} = \gamma {I} - \xi{R} $$)",
+           r"(Reproductive ratio $$R_0 =  \frac{\beta}{\gamma} S(0)^q$$)", #need to double check 
+           "q-Value $$\\begin{equation} x = \\begin{cases} 1 & \\text{, } frequency − dependent \\\\ 0 & \\text{, } density − dependent \\end{cases}\\end{equation} $$"
```
Comment (bryce-carson) [inline R/model_LaTeX.R:32]: Please use raw strings when writing $\LaTeX{}$. Inside the raw string you will only have to write $\LaTeX{}$, and won't need to escape backslash characters which are used to call macros and commands.

File: R/model_LaTeX.R [hunk 4]
```diff
@@ -26,25 +22,43 @@ SIR_LaTeX <- function(mu) {
...
+           ),
+         helpText(
+           # want to make sure it is \nu{S}
+           r"(Susceptible $$\frac{dS}{dt} = \mu{N}- \beta \frac{ S I}{N} + \xi{R} - \nu{S}$$)", 
+           r"(Infectious $$\frac{dI}{dt} = \frac{\beta S I}{N} - \gamma {I} - \nu{I}$$)",
+           r"(Recovered $$\frac{dR}{dt} = \gamma {I} - \xi{R} - \nu{R}$$)",
+           r"(Reproductive ratio $$R_0 =  \frac{\beta}{\gamma} S(0)^q$$)", #need to double check 
+           "q-Value $$\\begin{equation} x = \\begin{cases} 1 & \\text{, } frequency − dependent \\\\ 0 & \\text{, } density − dependent \\end{cases}\\end{equation} $$"
```
Comment (bryce-carson) [inline R/model_LaTeX.R:40]: Please use raw strings when writing $\LaTeX{}$. Inside the raw string you will only have to write $\LaTeX{}$, and won't need to escape backslash characters which are used to call macros and commands.

File: R/model_LaTeX.R [hunk 5]
```diff
@@ -26,25 +22,43 @@ SIR_LaTeX <- function(mu) {
...
       r"(Susceptible $$\frac{dS}{dt} = - \beta \frac{ S I}{N^q}$$)",
-      r"(Infectious $$\frac{dI}{dt} = \frac{\beta S I}{N^q} - \gamma I - \delta I $$)",
-      r"(Recovered $$\frac{dR}{dt} = \gamma I $$)",
-      r"(Dead $$\frac{dD}{dt} = \delta I $$)",
+      r"(Infectious $$\frac{dI}{dt} = \frac{\beta S I}{N^q} - \gamma {I} - \delta {I} $$)",
+      r"(Recovered $$\frac{dR}{dt} = \gamma {I} $$)",
+      r"(Dead $$\frac{dD}{dt} = \delta {I} $$)",
       r"(Reproductive ratio $$R_0 =  \frac{\beta}{\gamma + \delta} S(0)^q$$)",
-      "q-Value", br(), r"($$ 1, frequency-dependent $$)", br(), r"($$ 0, density-dependent $$)"
+      "q-Value $$\\begin{equation} x = \\begin{cases} 1 & \\text{, } frequency − dependent \\\\ 0 & \\text{, } density − dependent \\end{cases}\\end{equation} $$"
```
Comment (bryce-carson) [inline R/model_LaTeX.R:53]: Please use raw strings when writing $\LaTeX{}$. Inside the raw string you will only have to write $\LaTeX{}$, and won't need to escape backslash characters which are used to call macros and commands.

File: R/model_LaTeX.R [hunk 6]
```diff
@@ -26,25 +22,43 @@ SIR_LaTeX <- function(mu) {
...
       r"(Susceptible $$\frac{dS}{dt} =\mu_B N - \mu_D S - \beta \frac{ S I}{N^q}$$)",
-      r"(Infectious $$\frac{dI}{dt} = \frac{\beta S I}{N^q} - \gamma I - \delta I - \mu_D I$$)",
-      r"(Recovered $$\frac{dR}{dt} = \gamma I - \mu_D R $$)",
-      r"(Dead $$\frac{dD}{dt} = \delta I $$)",
+      r"(Infectious $$\frac{dI}{dt} = \frac{\beta S I}{N^q} - \gamma {I} - \delta {I} - \mu_D I$$)",
+      r"(Recovered $$\frac{dR}{dt} = \gamma {I} - \mu_D R $$)",
+      r"(Dead $$\frac{dD}{dt} = \delta {I} $$)",
       r"(Reproductive ratio $$R_0 =  \frac{\beta}{\gamma + \delta} S(0)^q$$)",
-      "q-Value", br(), r"($$ 1, frequency-dependent $$)", br(), r"($$ 0, density-dependent $$)"
+      "q-Value $$\\begin{equation} x = \\begin{cases} 1 & \\text{, } frequency − dependent \\\\ 0 & \\text{, } density − dependent \\end{cases}\\end{equation} $$"
```
Comment (bryce-carson) [inline R/model_LaTeX.R:61]: Please use raw strings when writing $\LaTeX{}$. Inside the raw string you will only have to write $\LaTeX{}$, and won't need to escape backslash characters which are used to call macros and commands.

File: R/model_LaTeX.R [hunk 7]
```diff
@@ -53,19 +67,19 @@ SEIR_LaTeX <- function(mu) {
...
       r"(Susceptible $$\frac{dS}{dt} = - \beta \frac{ S I}{N^q}$$)",
-      r"(Exposed $$\frac{dE}{dt} = \beta \frac{ S I}{N^q} - \gamma E $$)",
-      r"(Infectious $$\frac{dI}{dt} = \gamma E - \sigma I $$)",
-      r"(Recovered $$\frac{dR}{dt} = \sigma I $$)",
+      r"(Exposed $$\frac{dE}{dt} = \beta \frac{ S I}{N^q} - \gamma {E} $$)",
+      r"(Infectious $$\frac{dI}{dt} = \gamma {E} - \sigma {I} $$)",
+      r"(Recovered $$\frac{dR}{dt} = \sigma {I} $$)",
       r"(Reproductive ratio $$R_0 =  \frac{\beta}{\gamma}$$)",
-      "q-Value", br(), r"($$ 1, frequency-dependent $$)", br(), r"($$ 0, density-dependent $$)"
+      "q-Value $$\\begin{equation} x = \\begin{cases} 1 & \\text{, } frequency − dependent \\\\ 0 & \\text{, } density − dependent \\end{cases}\\end{equation} $$"
```
Comment (bryce-carson) [inline R/model_LaTeX.R:74]: Please use raw strings when writing $\LaTeX{}$. Inside the raw string you will only have to write $\LaTeX{}$, and won't need to escape backslash characters which are used to call macros and commands.

File: R/model_LaTeX.R [hunk 8]
```diff
@@ -53,19 +67,19 @@ SEIR_LaTeX <- function(mu) {
...
     ),
     helpText(
       r"(Susceptible $$\frac{dS}{dt} =\mu_B N - \mu_D S - \beta \frac{ S I}{N^q}$$)",
-      r"(Exposed $$\frac{dE}{dt} = \beta \frac{ S I}{N^q} - \gamma E - \mu_D E$$)",
+      r"(Exposed $$\frac{dE}{dt} = \beta \frac{ S I}{N^q} - \gamma {E} - \mu_D {E}$$)",
       r"(Infectious $$\frac{dI}{dt} = \gamma E - \sigma I - \mu_D I $$)",
       r"(Recovered $$\frac{dR}{dt} = \sigma I - \mu_D R $$)",
       r"(Reproductive ratio $$R_0 =  \frac{\beta}{\gamma}$$)",
-      "q-Value", br(), r"($$ 1, frequency-dependent $$)", br(), r"($$ 0, density-dependent $$)"
+      "q-Value $$\\begin{equation} x = \\begin{cases} 1 & \\text{, } frequency − dependent \\\\ 0 & \\text{, } density − dependent \\end{cases}\\end{equation} $$"
```
Comment (bryce-carson) [inline R/model_LaTeX.R:82]: Please use raw strings when writing $\LaTeX{}$. Inside the raw string you will only have to write $\LaTeX{}$, and won't need to escape backslash characters which are used to call macros and commands.

File: R/model_LaTeX.R [hunk 9]
```diff
@@ -79,7 +93,7 @@ SEIRD_LaTeX <- function(mu) {
       r"(Recovered $$\frac{dR}{dt} = \sigma I $$)",
       r"(Dead $$ \frac{dD}{dt} = \delta I $$)",
       r"(Reproductive ratio $$R_0 =  \frac{\beta}{\gamma}$$)",
-      "q-Value", br(), r"($$ 1, frequency-dependent $$)", br(), r"($$ 0, density-dependent $$)"
+      "q-Value $$\\begin{equation} x = \\begin{cases} 1 & \\text{, } frequency − dependent \\\\ 0 & \\text{, } density − dependent \\end{cases}\\end{equation} $$"
```
Comment (bryce-carson) [inline R/model_LaTeX.R:96]: Please use raw strings when writing $\LaTeX{}$. Inside the raw string you will only have to write $\LaTeX{}$, and won't need to escape backslash characters which are used to call macros and commands.

File: R/model_LaTeX.R [hunk 10]
```diff
@@ -88,7 +102,7 @@ SEIRD_LaTeX <- function(mu) {
       r"(Recovered $$\frac{dR}{dt} = \sigma I - \mu_D R $$)",
       r"(Dead $$ \frac{dD}{dt} = \delta I $$)",
       r"(Reproductive ratio $$R_0 =  \frac{\beta}{\gamma}$$)",
-      "q-Value", br(), r"($$ 1, frequency-dependent $$)", br(), r"($$ 0, density-dependent $$)"
+      "q-Value $$\\begin{equation} x = \\begin{cases} 1 & \\text{, } frequency − dependent \\\\ 0 & \\text{, } density − dependent \\end{cases}\\end{equation} $$"
```
Comment (bryce-carson) [inline R/model_LaTeX.R:105]: Please use raw strings when writing $\LaTeX{}$. Inside the raw string you will only have to write $\LaTeX{}$, and won't need to escape backslash characters which are used to call macros and commands.

File: episimFork.Rproj
Comment (bryce-carson) [inline episimFork.Rproj:1]: Please ammend the commit and unstage this file.

**Review by bryce-carson — CHANGES_REQUESTED** [review]
- There are some requested changes needed to make the $\LaTeX{}$ readable.
- Please amend the commit and remove the Rproj file.

**Comment by kle6951** [comment]
I have made the changes. Please let me know if anything needs to be fixed.

File: episimFork.Rproj
Comment (bryce-carson) [reply episimFork.Rproj:1]: Don't worry about this anymore.

**Comment by bryce-carson** [comment]
> I have made the changes. Please let me know if anything needs to be fixed.

I don't see any new commits to this branch. Where did you make the changes?

**Comment by kle6951** [comment]
> > I have made the changes. Please let me know if anything needs to be fixed.
> 
> I don't see any new commits to this branch. Where did you make the changes?

Are you able to see that I made changes by using the r"()" for the LaTex? I committed the changes already. If you don't see it, I will commit again

**Comment by bryce-carson** [comment]
> > > I have made the changes. Please let me know if anything needs to be fixed.
> > 
> > 
> > I don't see any new commits to this branch. Where did you make the changes?
> 
> Are you able to see that I made changes by using the r"()" for the LaTex? I committed the changes already. If you don't see it, I will commit again

No, I don't see the changes here. You can look at the commits for this PR by selecting the Commits tab in the tab bar below the PR title. That only shows the first commit you made to this PR, and then a commit I made which resolves a merge conflict.

Make sure you check out this branch (`feat-LaTex`, note the casing) and then make the commit and push to the remote branch.

**Comment by kle6951** [comment]
I committed to the branch already. Please review it.

**Comment by bryce-carson** [comment]
> I committed to the branch already. Please review it.

The commit to the branch is visible now. Thanks.

**Pull request merged**